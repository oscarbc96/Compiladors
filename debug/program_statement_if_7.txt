FLEX: Detected mode PRGM
BISON: Starting in mode PRGM
FLEX: ASSIGN yytex: :=
FLEX: FLOAT yytex: 20.0 -> 20.000000
FLEX: IF yytex: if
BISON: Checking mode
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: COMP yytex: > -> >
FLEX: INT yytex: 10 -> 10
FLEX: AND yytex: and
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: COMP yytex: < -> <
FLEX: INT yytex: 20 -> 20
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: OR yytex: or
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: COMP yytex: = -> =
FLEX: INT yytex: 444 -> 444
FLEX: AND yytex: and
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: COMP yytex: <> -> <>
FLEX: INT yytex: 555 -> 555
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: THEN yytex: then
FLEX: INT yytex: 123 -> 123
FLEX: INT yytex: 123 -> 123
FLEX: INT yytex: 123 -> 123
FLEX: FI yytex: fi
The compilation has been successful
