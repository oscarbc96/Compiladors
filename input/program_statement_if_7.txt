calc off

a := 20.0

if ((a > 10 and a < 20) or (a = 444 and a <> 555)) then 
  123
  123
  123
fi
