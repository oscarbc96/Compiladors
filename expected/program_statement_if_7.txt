1: $t01 := 20.000000
2: a := $t01
3: IF a GTF 10 GOTO 5
4: GOTO 7
5: IF a LTF 20 GOTO 11
6: GOTO 7
7: IF a EQ 444 GOTO 9
8: GOTO 17
9: IF a NE 555 GOTO 11
10: GOTO 17
11: PARAM 123
12: CALL PUTI,1
13: PARAM 123
14: CALL PUTI,1
15: PARAM 123
16: CALL PUTI,1
17: HALT
